docs/user_guide/plotting.md

```diff
@@ -1,11 +1,341 @@
+:width: 100%
+:::
+
+:::{figure} ../plots/cf.PlotVariables1D_tpl_config_analy__1__c80529af83__plot__proc_2_a2211e799f__cat_incl__var_n_jet__c2.pdf
+:width: 100%
+:::
+::::
+
+
+
+:::{dropdown} Limitations of the ```variable_settings```
+While in theory, we can change anything inside the variable and process instances via the
+```variable_settings```, there are certain attributes that are already used during the creation
+of the histograms (e.g. the ```expression``` and the ```binning```) and since our ```variable_settings```
+parameter only modifies these attributes during the runtime of our plotting task, this will not
+impact our final results.
+:::
+
+
+For the ```general_settings```, ```process_settings```, and ```variable_settings``` you can define
+defaults and groups in the config, e.g. via
+
+```python
+config_inst.x.default_variable_settings = {"jet1_pt": {"rebin": 4, "x_title": r"Leading jet $p_{T}$"}}
+config_inst.x.process_settings_groups = {
+    "unstack_processes": {proc: {"unstack": True} for proc in ("tt", "st")},
+}
+config_inst.x.general_settings_groups = {
+    "compare_shapes": {"skip_ratio": True, "shape_norm": True, "yscale": "log", "cms_label": "simpw"},
+}
+```
+The default is automatically used when no parameter is given in the task call, and the groups can
+be used directly on the command line and will be resolved automatically. Our previously defined
+defaults and groups will be used e.g. by the following task call:
+
+```shell
+law run cf.PlotVariables1D --version v1 --processes tt,st --variables n_jet,jet1_pt \
+    --process-settings unstack_processes --general-settings compare_shapes
+```
+
+::::{grid} 1 1 2 2
+:::{figure} ../plots/cf.PlotVariables1D_tpl_config_analy__1__be60d3bca7__plot__proc_2_a2211e799f__cat_incl__var_jet1_pt__c3.pdf
+:width: 100%
+:::
+
+:::{figure} ../plots/cf.PlotVariables1D_tpl_config_analy__1__be60d3bca7__plot__proc_2_a2211e799f__cat_incl__var_n_jet__c3.pdf
+:width: 100%
+:::
+::::
+
+
+## Creating 2D plots
+
+Columnflow also provides the {py:class}`~columnflow.tasks.plotting.PlotVariables2D` to create
+two-dimensional plots. Two-dimensional histograms are created by passing two variables to the
+```--variables``` parameter, separated by a ```-```. Here is an exemplary task call and their
+outputs.
+```shell
+law run cf.PlotVariables2D --version v1 \
+    --processes tt,st --variables n_jet-jet1_pt,jet1_pt-n_jet
+```
+
+::::{grid} 1 1 2 2
+:::{figure} ../plots/cf.PlotVariables2D_tpl_config_analy__1__b27b994979__plot__proc_2_a2211e799f__cat_incl__var_jet1_pt-n_jet.pdf
+:width: 100%
+:::
+
+:::{figure} ../plots/cf.PlotVariables2D_tpl_config_analy__1__b27b994979__plot__proc_2_a2211e799f__cat_incl__var_n_jet-jet1_pt.pdf
+:width: 100%
+:::
+::::
+
+For more information on the task parameters of the ```PlotVariables2D``` task, take a look into the
+[plotting task overview](../task_overview/plotting.md).
```

Comment:
Maybe mention here that many (if not all!) options used in PlotVariables1D can also be used for the 2D plots